test/sql/types/decimal/test_decimal.test

```diff
@@ -114,6 +114,17 @@ select '0.00003'::DECIMAL(38,10)::VARCHAR
 ----
 0.0000300000
 
+# Downcasting should round
+query I
+select CAST(33.846 AS DECIMAL(5,0)) d0;
```

Comment:
Can we add a test when the rounding results in an overflow for the decimal type, e.g. in Postgres:

```sql
select cast('9.99' as decimal(1,0));
--- ERROR:  numeric field overflow
--- DETAIL:  A field with precision 1, scale 0 must round to an absolute value less than 10^1.

```

Reply:
Turns out this doesn't hit the new code.

Reply:
Have looked at this again - the reason it doesn't hit the new code is because there are two decimal operators - `DecimalScaleDownOperator` and `DecimalScaleDownCheckOperator`. Currently the implementation only exists for `DecimalScaleDownOperator` - which means the rounding will only happen when `source_width >= target_width`, i.e. when going down in precision no rounding happens. Perhaps this can be extended to work in both cases?

Reply:
It's in the string casting code actually. That does wasn't checking whether rounding overflowed.